main.yml

```diff
@@ -0,0 +1,42 @@
+name: build
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: [3.2, 3.3, 3.4, 3.5]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Use H3 ${{ matrix.GO }}
+    
+  report-generation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Use v3.4.0 
+        uses: actions/setup-go@v1
+        with: v3.4.0
```

Comment:
like above job, you want the golang version, not H3 version here

Reply:
reuse `${{ matrix.GO }}` is an idea, but im not sure how that will play with the "upload" step that follows.  perhaps just the latest go version is good enough?